azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachineScaleSet.java

```diff
@@ -221,6 +222,19 @@
     Map<String, VirtualMachineScaleSetExtension> extensions();
 
     /**
+     * @return the network interfaces associated with all virtual machine instances in a scale set
+     */
+    PagedList<VirtualMachineScaleSetNetworkInterface> listNetworkInterfaces();
```

Comment:
There are some advantage returning as  `PagedList` when the inner collection actually returns `PagedList`.

An advanced users can make use of `PagedList::currentPage()` `PagedList::hasNextPages()` `PagedList::nextPage()` and `PagedList::loadNextPage()` methods. 

One use case is - suppose a ModelA has a collection property of type ModelB. Now given a `PagedList` of ModelA user want to build a single `LAZY` paged list of ModelB by flattening the ModelB collection property in each ModelA instance. In order to build a fully `LAZY` list of ModelB, access to above mentioned `PagedList` methods of ModelA can be used.

My understanding is - from our fluent interface we return `List` only if the underlying collection is not a `PagedList`. Our core interfaces such as `SupportsListingByGroup`, `SupportsListingByRegion` are already declared to return `PagedList<T>`

```java
public interface SupportsListingByGroup<T> {
    PagedList<T> listByGroup(String resourceGroupName);
}
```

Also, `PagedList` is of of type  `List`, so users also treat it as regular list. 

Please share your thoughts ..